packages/client/src/monaco-vscode-api-services.ts

```diff
@@ -0,0 +1,162 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2018-2022 TypeFox GmbH (http://www.typefox.io). All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { editor, Uri } from 'monaco-editor/esm/vs/editor/editor.api.js';
+import { initialize as initializeMonacoService } from 'vscode/services';
+import { initialize as initializeVscodeExtensions } from 'vscode/extensions';
+import type { OpenEditor } from 'vscode/service-override/modelEditor';
+
+export type InitializeServiceConfig = {
+    enableDialogService?: boolean;
+    enableNotificationService?: boolean;
+    enableModelEditorService?: boolean;
+    modelEditorServiceConfig?: {
+        useDefaultFunction: boolean;
+        openEditorFunc?: OpenEditor
+    };
+    enableConfigurationService?: boolean
+    configurationServiceConfig?: {
+        defaultWorkspaceUri: string;
+    };
+    enableThemeService?: boolean;
+    enableKeybindingsService?: boolean;
+    enableTextmateService?: boolean;
+    enableLanguagesService?: boolean;
+    enableAudioCueService?: boolean;
+    enableDebugService?: boolean;
+    enablePreferencesService?: boolean;
+    userServices?: editor.IEditorOverrideServices;
+    debugLogging?: boolean;
+};
+
+export const initServices = async (config?: InitializeServiceConfig) => {
+    await importAllServices(config)
+        .then(() => {
+            if (config?.debugLogging === true) {
+                console.log('initializeMonacoService completed successfully');
+            }
+        })
+        .then(async () => await initializeVscodeExtensions())
+        .then(() => {
+            if (config?.debugLogging === true) {
+                console.log('initializeVscodeExtensions completed successfully');
+            }
+        })
+        .catch((e: Error) => { throw e; });
```

Comment:
What about using async/await?

Reply:
You mean just write await statements and not intercept the Error here an let the next level deal with it catch it?
```Typescript
export const initServices = async (config?: InitializeServiceConfig) => {
    await importAllServices(config);
    if (config?.debugLogging === true) {
        console.log('initializeMonacoService completed successfully');
    }
    await initializeVscodeExtensions();
    if (config?.debugLogging === true) {
        console.log('initializeVscodeExtensions completed successfully');
    }
};
```

Reply:
My comment doesn't concern the error management, it is about not using `.then` or `.catch`

btw, `.catch((e: Error) => { throw e; });` seems to be pointless, isn't it?

Reply:
> My comment doesn't concern the error management, it is about not using .then or .catch

Yes, I got this. I wanted to chain the catch, but using a catch is fine, but here it does not even make sense, because what was there is pointless.

Reply:
Changed as suggested. Remove superfluous catch/re-throw